tests/rules/test_terraform_no_command.py

```diff
@@ -0,0 +1,27 @@
+import pytest
+from thefuck.rules.terraform_no_command import match, get_new_command
+from thefuck.types import Command
+
+
+@pytest.mark.parametrize('script, output', [
+    ('terraform appyl', 'Terraform has no command named \"appyl\". Did you mean \"apply\"?)'),
+    ('terraform destory', 'Terraform has no command named \"destory\". Did you mean \"destroy\"?)')])
+def test_match(script, output):
+    assert match(Command(script, output))
+
+
+@pytest.mark.parametrize('script, output', [
+    ('terraform --version', 'Terraform v0.12.2'),
+    ('terraform plan', 'No changes. Infrastructure is up-to-date.'),
+    ('terraform apply', 'Apply complete! Resources: 0 added, 0 changed, 0 destroyed.'),
+])
+def test_not_match(script, output):
+    assert not match(Command(script, output=output))
+
+
+@pytest.mark.parametrize('command, new_command', [
+    (Command('terraform appyl', 'Terraform has no command named \"appyl\". Did you mean \"apply\"?)'), 'terraform apply'),
+    (Command('terraform destory --some-other-option', 'Terraform has no command named \"destory\". Did you mean \"destroy\"?)'), 'terraform destroy --some-other-option'),
+])
+def test_get_new_command(command, new_command):
+    assert get_new_command(command) == new_command
```

Comment:
I like that you passed in string parameters and instantiated `Command` with them. How about doing the same here?

```suggestion
@pytest.mark.parametrize('script, output, new_command', [
    ('terraform appyl', 'Terraform has no command named "appyl". Did you mean "apply"?', 'terraform apply',),
    ('terraform destory --some-other-option', 'Terraform has no command named "destory". Did you mean "destroy"?', 'terraform destroy --some-other-option',),
])
def test_get_new_command(script, output, new_command):
    assert get_new_command(Command(script, output)) == new_command
```